### DISCUSSION TITLE
How to customize lazygit to my needs (Please follow the description)?

### DISCUSSION THREAD

**Discussion by Palash90** [jesseduffield/lazygit#2202] [Ideas]
I am trying to mimic an ide using vim. I have made some progress with splits and panes.

Following is my .vimrc file.
`
set number
set ruler
set smartindent
"set tw=60
set shellcmdflag=-ic
set nocp
"Take care of indents for Java.
set autoindent
set si
set shiftwidth=4
set cinoptions+=j1
set nocompatible
syntax on
set cindent 
set tabstop=4
set expandtab
filetype indent on
let python_highlight_all = 1
let g:netrw_banner = 0
let g:netrw_liststyle = 3
let g:netrw_browse_split = 4
let g:netrw_altv = 1
let g:netrw_winsize = 20
augroup ProjectDrawer
    autocmd!
    autocmd VimEnter * :Vexplore
    autocmd VimEnter * wincmd l
    autocmd VimEnter * :term
    autocmd VimEnter * wincmd j
    autocmd VimEnter * :resize +10
    autocmd VimEnter * wincmd x
    autocmd VimEnter * wincmd k
    autocmd VimEnter * :vert term htop -p 999999
    autocmd VimEnter * wincmd x
    autocmd VimEnter * wincmd h
    autocmd VimEnter * :vertical resize -15
    autocmd VimEnter * wincmd l
    autocmd VimEnter * :vertical resize +20
augroup END
`
With this, it looks like the following.
![Screenshot from 2022-10-05 12-22-53](https://user-images.githubusercontent.com/5032317/193998763-ef6af6d4-15f7-4f02-b960-301d35ad6f51.png)

In this screen, there is htop running on the top right corner. I want that to be replaced with lazygit left pane. And when I select a file from the file explorer pane, the diff should open in the main editor pane (The middle pane).

Is it possible to achieve this?

**Comment by mark2185** [comment]
Highly unlikely, you can tell `lazygit` to open up the file under the cursor in a specific `vim` process by starting `vim` as a `server`, but I'm not sure you can specify the window.

Ultimately, I don't see the point, there are `vim` plugins for `git` interaction, such as [vim-fugitive](https://github.com/tpope/vim-fugitive), but you can also launch `lazygit` in a terminal in a new tab and then open the files from there, in a new `vim` process. A bit nested, but if that's what you want...

I have this for that purpose:
```
command! Lazygit tab term ++close lazygit
nnoremap <leader>T :Lazygit<CR>
```

In short, maybe it's better _not_ to try and make `vim` a "classic" IDE.
